### ISSUE TITLE
Accessibility Issues-Keyboard Navigation to 'Get Directions' Not Possible

### ISSUE THREAD

**Issue by bkozik** [CodeCabin/wp-google-maps#1081]
The plugin seems to support keyboard navigation in general, I am able to move between markers etc. 

However, I am not able to get focus focus/read the 'Get Directions' button (or the 'x') when a marker is selected and the directions-box pops up. I believe this should be supported functionality. 

Let me know if you need any more information!

**Comment by DylanCodeCabin** [comment]
Hi @bkozik, 

Thank you for reaching out, we do appreciate your time on this one. Most of the in-map keyboard control is handled directly by the mapping engine (Google Maps in most cases), meaning we don't directly handle keyboard controls. 

With that said, we do implement tab-indexing which is to say that tapping tab can help navigate through some of the in-map elements thanks to the way that these are handled by most browsers. 

I did run a few tests and I was able to get to the Get Directions link with tab controls, however the X does not seem to be directly accessible via this method. That X is directly from the Google Maps engine, which means we don't directly control that element. 

With that said, we'd be open to working on this more in the future, to make keyboard navigation more user friendly, as we definitely see how that would be helpful to users. 

Would you be open to sharing a short video of this with me? Perhaps even highlighting the keyboard keys you are currently using, and perhaps how you imagine it could be improved. 

That would allow me to sit down with the rest of my team so that we can look into how we can work towards that in the future. Additionally, depending on what you need, we might be able to help with a short term solution for you in the meantime as well.

**Comment by bkozik** [comment]
Thanks for the prompt response @DylanCodeCabin !

From what I can tell, the 'x' has `aria-hidden` on it, which might be part of the problem. When using Apple's VoiceOver, I am able to navigate with keyboard commands to the 'Get Directions', but when it's not active 'tab' simply finishes going through the markers and on to the next element in the site. 

I've gotten a temporary workaround going with custom js/css for focus indicators and tabindexes.

 I'm very curious about the difference between my setup and yours such that you were able to tab to get directions. Let me know what information you might need about my configuration! There are no tabindexes until I manually added them with the custom js?

**Comment by DylanCodeCabin** [comment]
My pleasure @bkozik - Yes, the 'X' isn't an element that we create, instead it's created directly by the Google Maps API, meaning we don't directly attribute it with anything, although that's something we will expand on when we work on our accesibility roadmap item listed here: https://trello.com/c/EE8VHpfs/50-wave-ada-wcag 

That's still some time out in terms of our production schedule, but accessibility in general is something we are eager on improving dramatically in the future. 

I'm happy to hear you've managed to get a temporary solution working on your side in the meantime though. 

I've just checked our core code and you are correct, we don't tabindex the get directions control at the moment, so that's definitely something we need to address. In my environment I am using Chrome on desktop, which might be another reason why we are seeing differences. 

I was able to open the info-window on my side by tabbing to a marker and once the info window was open tabbing two more times got me to the Get Directions link, because I had a sample link in one of my info-window descriptions. I don't have enough data on hand to say whether or not this is something Chrome handled internally, tracing back to that link, or if this is something that the Google Maps API polyfills for my environment. 

With that said, you can expect to see big changes to the way we handle this in the future, as I expect we'll be doing the following as a minimum: 
- Rebuilding the structures to be inherently accessible, to a few standards (WAVE, ADA, WCAG)
- Add more content filters allowing more advanced users to restructure HTML to their preference, which will likely evolve into custom template options 
- And now that I've spoken to you about this, we'll likely add even more emphasis on internal keyboard supports as well 

I don't have a timeline for these just yet, but they are slated for near future.

**Comment by DylanCodeCabin** [comment]
We have reached our first phase of compliance, although we do have larger plans in the future. 

For the time being, the following document serves as a good resource for how we've approached this: https://www.wpgmaps.com/help/docs/accessibility-and-wcag-compliance/

We're also opening this up to community feedback for the long-term.
